src/github.com/couchbase/sync_gateway/db/database.go

```diff
@@ -197,6 +201,8 @@ func (dc *DatabaseContext) TakeDbOffline() error {
 
 func (context *DatabaseContext) Authenticator() *auth.Authenticator {
 	// Authenticators are lightweight & stateless, so it's OK to return a new one every time
+	context.BucketLock.RLock()
+	defer context.BucketLock.RUnlock()
```

Comment:
Why does only the Authenticator need the bucket read lock handling here, and not other bucket operations?